gocourse10/feed/feed.go

```diff
@@ -0,0 +1,59 @@
+package feed
+
+import (
+	"log"
+)
+
+func Feed(c Collar) {
+	log.Printf("Feed with %s \n", c.Trough.Food())
+}
+
+type Trough interface {
+	Food() string
+}
+
+type MeetTrought struct {
+}
+
+func (MeetTrought) Food() string {
+	return "Ham, pork and chicken"
+}
+
+type VegetablesTrought struct {
+}
+
+func (VegetablesTrought) Food() string {
+	return "Carrots, potatoes and cucumbers"
+}
+
+type FishTrough struct {
+}
+
+func (FishTrough) Food() string {
+	return "Code and salmon"
+}
+
+type Collar struct {
+	Trough
+	ID int
```

Comment:
Для інкапсуляції:
```suggestion
	trough Trough
	id     int
```

І додати конструктор `NewCollar`.